### PR TITLE
Fix TestTemporaryToFunction

### LINKED ISSUE #777: rework dawn unit test: dawn/Optimizer/Passes/TestTemporaryToFunction
seems to do nothing. Add a test here

### PR DESCRIPTION
**Pull request by mroethlin**
## Technical Description

The TestTemporaryToFunction test (Optimizer/Passes) now does something useful. Two iir files have been introduced, one that prompts a promotion to a function, and one that doesn't. 

### Resolves / Enhances

Fixes #777 

### Note

It is currently not clear how we can update / maintain the reference iirs. 

### Dependencies

This PR is independent

### CODE REVIEW

**Comment by mroethlin** [comment]
launch jenkins

File: dawn/src/dawn/Optimizer/PassTemporaryToStencilFunction.cpp [hunk 1]
```diff
@@ -767,6 +769,6 @@ bool PassTemporaryToStencilFunction::run(
   }
 
   return true;
-}
+} // namespace dawn
```
Comment (twicki) [inline dawn/src/dawn/Optimizer/PassTemporaryToStencilFunction.cpp]: this was probably due to a parenthesis mismatch at some point, should not be here

File: dawn/test/unit-test/dawn/Optimizer/Passes/TestTemporaryToFunction.cpp [hunk 2]
```diff
@@ -12,54 +12,95 @@
...
+    const auto& stencil = *instantiation->getIIR()->getChildren().front();
 
-    auto stencilInstantiationMap = compiler_.optimize(compiler_.lowerToIIR(sir));
-    // Report diagnostics
-    if(compiler_.getDiagnostics().hasDiags()) {
-      for(const auto& diag : compiler_.getDiagnostics().getQueue())
-        std::cerr << "Compilation Error " << diag->getMessage() << std::endl;
-      throw std::runtime_error("compilation failed");
+    // temp to function pass expects stmt data to contain the stack trace (can be empty though)
+    for(const auto& multiStage : stencil.getChildren()) {
```
Comment (twicki) [inline dawn/test/unit-test/dawn/Optimizer/Passes/TestTemporaryToFunction.cpp]: can we collapse this into one loop?

Comment (mroethlin) [reply dawn/test/unit-test/dawn/Optimizer/Passes/TestTemporaryToFunction.cpp]: I'm not sure and its not the first time I have been wondering how to achieve this using our code base. I could obviously do it by stringing together this Frankensteinian construct:
```
for(const auto& stmt : stencil.getChildren()
                               .front()
                               ->getChildren()
                               .front()
                               ->getChildren()
                               .front()
                               ->getAST()
                               .getStatements()) {
      stmt->getData<iir::IIRStmtData>().StackTrace = std::vector<dawn::ast::StencilCall*>();
    }
```
But is this really what you/we want? Maybe a utility class blindly stringing together all statements in a stencilInstantiation is in order?

**Comment by mroethlin** [comment]
launch jenkins

File: dawn/test/unit-test/dawn/Optimizer/Passes/TestTemporaryToFunction.cpp (hunk 2 above)
Comment (twicki) [reply dawn/test/unit-test/dawn/Optimizer/Passes/TestTemporaryToFunction.cpp]: that is why we have IterateIIROver

**Comment by mroethlin** [comment]
launch jenkins

**Pull request merged**